runtime/parser/type.go

```diff
@@ -544,10 +545,36 @@ func defineRestrictedOrDictionaryType() {
 	)
 }
 
-// parseNominalTypes parses zero or more nominal types separated by comma.
+func parseNominalType(
+	p *parser,
+	rightBindingPower int,
+	rejectAccessKeywords bool,
+) (*ast.NominalType, error) {
+	ty, err := parseType(p, lowestBindingPower)
+	if err != nil {
+		return nil, err
+	}
+	nominalType, ok := ty.(*ast.NominalType)
+	if !ok {
+		return nil, p.syntaxError("unexpected non-nominal type: %s", ty)
+	}
+	if rejectAccessKeywords &&
+		nominalType.Identifier.Identifier == KeywordAll ||
+		nominalType.Identifier.Identifier == KeywordAccess ||
+		nominalType.Identifier.Identifier == KeywordAccount ||
+		nominalType.Identifier.Identifier == KeywordSelf {
+		return nil, p.syntaxError("unexpected non-nominal type: %s", ty)
+	}
+	return nominalType, nil
```

Comment:
maybe use a switch-case here:
```suggestion
	if rejectAccessKeywords {
		switch nominalType.Identifier.Identifier {
		case KeywordAll, KeywordAccess, KeywordAccount, KeywordSelf
			return nil, p.syntaxError("unexpected non-nominal type: %s", ty)
		}
	}
	return nominalType, nil
```